syntax = "proto3";

package pb;

message data_req
{
    optional bytes name = 1;        // name of the file
    optional bytes data = 2;         // bytes of the file
    optional uint32 perms = 3;
    optional uint64 id = 4;          // for debugging
}

message data_res
{
    optional bool success = 1;
    optional uint64 id = 2;          // for debugging
}

message done_req
{
    optional bytes name = 1;        // name of the last file transferred
    optional uint64 id = 2;          // for debugging
}

message done_res
{
    optional bool success = 1;
    optional uint64 id = 2;          // for debugging
}

service service
{
    rpc data_transfer(data_req) returns (data_res);
    rpc done_transfer(done_req) returns (done_res);
}

option cc_generic_services = true;
